flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/plan/rules/logical/ExpressionReductionRulesTest.scala

```diff
@@ -46,4 +47,21 @@ class ExpressionReductionRulesTest extends TableTestBase {
     util.verifyPlan("SELECT myUdf(1) FROM MyTable")
   }
 
+  @Test
+  def testExpressionReductionWithPythonUDF(): Unit = {
+    util.addFunction("PyUdf", MockedPythonUDFWithoutArguments)
+    util.addFunction("MyUdf", Func1)
+    util.verifyPlan("SELECT PyUdf(), MyUdf(1) FROM MyTable")
+  }
+}
+
+object MockedPythonUDFWithoutArguments extends ScalarFunction {
```

Comment:
Check both deterministic and non-deterministic python udfs?

Reply:
IMO it's not necessary to check non-deterministic python UDFs in this rule test. My reason is:
1. If the UDF is non-deterministic, it won't match this rule whether it is a Python UDF or not. This test case won't cover more code.
2. We already have a ITCase about non-deterministic python UDF in `test_udf.py` to make sure the non-deterministic python UDFs work properly.